ai-platform-unified/notebooks/official/featurestore_movie_prediction.ipynb

```diff
@@ -0,0 +1,1372 @@
+        "from google.cloud.aiplatform_v1beta1.types import \\\n",
+        "    featurestore_online_service as featurestore_online_service_pb2\n",
+        "from google.cloud.aiplatform_v1beta1.types import \\\n",
+        "    featurestore_service as featurestore_service_pb2\n",
+        "from google.cloud.aiplatform_v1beta1.types import io as io_pb2\n",
+        "from google.protobuf.duration_pb2 import Duration\n",
+        "\n",
+        "# Create admin_client for CRUD and data_client for reading feature values.\n",
+        "admin_client = FeaturestoreServiceClient(client_options={\"api_endpoint\": API_ENDPOINT})\n",
+        "data_client = FeaturestoreOnlineServingServiceClient(\n",
+        "    client_options={\"api_endpoint\": API_ENDPOINT}\n",
+        ")\n",
+        "\n",
+        "# Represents featurestore resource path.\n",
+        "BASE_RESOURCE_PATH = admin_client.common_location_path(PROJECT_ID, LOCATION)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "h_HmF24mBHv9"
+      },
+      "source": [
+        "## Terminology and Concept\n",
+        "\n",
+        "### Featurestore Data model\n",
+        "\n",
+        "Feature store organizes data with the following 3 important hierarchical concepts:\n",
+        "```\n",
+        "Featurestore -> EntityType -> Feature\n",
+        "```\n",
+        "* **Featurestore**: the place to store your features\n",
+        "* **EntityType**: under a Featurestore, an *EntityType* describes an object to be modeled, real one or virtual one.\n",
+        "* **Feature**: under an EntityType, a *feature* describes an attribute of the EntityType\n",
+        "\n",
+        "In the movie prediction example, you will create a feature store called *movie_prediction*. This store has 2 entity types: *Users* and *Movies*. The Users entity has age, gender, liked genres features. The Movies entity has genres, average rating features.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9UvxYyGUimKw"
+      },
+      "source": [
+        "## Create Featurestore and Define Schemas"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "buQBIv3ZL3A0"
+      },
+      "source": [
+        "### Create Featurestore\n",
+        "\n",
+        "The method to create a featurestore returns a\n",
+        "[long-running operation](https://google.aip.dev/151) (LRO). An LRO starts an asynchronous job. LROs are returned for other API\n",
+        "methods too, such as updating or deleting a featurestore. Calling\n",
+        "`create_fs_lro.result()` waits for the LRO to complete."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "FscHZa0DXjmC"
+      },
+      "outputs": [],
+      "source": [
+        "FEATURESTORE_ID = \"movie_prediction_{timestamp}\".format(timestamp=TIMESTAMP)\n",
```

Comment:
What is the reason for having timestamp as postfix? I wonder if it led people think FS is created dynamically.

Reply:
This is to temporarily give it a unique name, because if for some reason the tests failed, the created featurestore will not be cleaned up. Next time we run the test, we will get a "featurestore already exist" error.

This can be removed after we pass the test and merge it with upstream, when we will no longer need to run it in the test env.